### PR TITLE
Nodes: Fix VideoTexture in WebGPUBackend & ColorSpaceNode revision

### LINKED ISSUE #26233: WebGPURenderer produces washed out colours compared to WebGL
### Description

I've integrated a bundle of the WebGPURenderer to begin testing and I am seeing a difference in the colors produced compared to WebGLRenderer. They are highly exposed. I'm not sure if a property is needed for it. 

The WebXR manager is also not available yet and might need integration. 

### Reproduction steps

live examples

### Code

```
camera = new THREE.PerspectiveCamera( 75, window.innerWidth / window.innerHeight, 1, 1100 );

scene = new THREE.Scene();

const geometry = new THREE.SphereGeometry( 500, 60, 40 );
// invert the geometry on the x-axis so that all of the faces point inward
geometry.scale( - 1, 1, 1 );

const video = document.getElementById( 'video' );
video.play();

const texture = new THREE.VideoTexture( video );
texture.colorSpace = THREE.SRGBColorSpace;
const material = new THREE.MeshBasicMaterial( { map: texture } );

const mesh = new THREE.Mesh( geometry, material );
scene.add( mesh );

if (await WebGPU.isAvailable()) {
	renderer = new WebGPURenderer();
} else {
	renderer = new THREE.WebGLRenderer();
}



renderer.setPixelRatio( window.devicePixelRatio );
renderer.setSize( window.innerWidth, window.innerHeight );
container.appendChild( renderer.domElement );
```

### Live example

* WebGPU - https://danrossi.github.io/three-webgpu-renderer/tests/webgpu_video_panorama_equirectangular.html
* WebGL - https://danrossi.github.io/three-webgpu-renderer/tests/webgl_video_panorama_equirectangular.html

### Screenshots

_No response_

### Version

153dev

### Device

_No response_

### Browser

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by sunag**
Fixes: https://github.com/mrdoob/three.js/issues/26233

**Description**

Related: https://github.com/mrdoob/three.js/issues/26233#issuecomment-1589701409

### CODE REVIEW

File: examples/jsm/nodes/accessors/TextureNode.js [hunk 1]
```diff
@@ -136,7 +138,15 @@ class TextureNode extends UniformNode {
 
 			}
 
-			return builder.format( propertyName, nodeType, output );
+			let snippet = propertyName;
+
+			if ( builder.needsColorSpace( this.value ) ) {
```
Comment (LeviPesin) [inline examples/jsm/nodes/accessors/TextureNode.js:143]: I think we should name this method something like `needsForceLinear` or `needsConversionToLinear`?

File: examples/jsm/nodes/display/ColorSpaceNode.js [hunk 2]
```diff
@@ -26,82 +35,75 @@ export const LinearTosRGB = new ShaderNode( ( inputs ) => {
...
+
+const Methods = {
+	[ ColorSpaceNode.LINEAR_TO_sRGB ]: LinearTosRGBShader,
+	[ ColorSpaceNode.sRGB_TO_LINEAR ]: sRGBToLinearShader
+};
 
 export default ColorSpaceNode;
 
-export const colorSpace = ( node, colorSpace ) => nodeObject( new ColorSpaceNode( null, nodeObject( node ) ).fromColorSpace( colorSpace ) );
+export const linearToColorSpace = ( node, colorSpace ) => nodeObject( new ColorSpaceNode( getMethod( LinearSRGBColorSpace, colorSpace ), nodeObject( node ) ) );
```
Comment (LeviPesin) [inline examples/jsm/nodes/display/ColorSpaceNode.js:98]: Maybe just `toColorSpace` and `fromColorSpace`?

File: examples/jsm/nodes/accessors/TextureNode.js (hunk 1 above)
Comment (sunag) [reply examples/jsm/nodes/accessors/TextureNode.js:143]: Hmm.. Maybe needsColorConversionToLinear

File: examples/jsm/nodes/display/ColorSpaceNode.js (hunk 2 above)
Comment (sunag) [reply examples/jsm/nodes/display/ColorSpaceNode.js:98]: I think it's is not possible currently. We need to know what the source is to made the conversion properly.

**Pull request merged**